### PR TITLE
Align group hooks with GitLab API

### PR DESCRIPTION
**Pull request by jtymes**
`push_events_branch_filter` is not yet documented, but it is there. https://gitlab.com/gitlab-org/gitlab/-/merge_requests/82531

`subgroup_events` has been around since 2021 https://gitlab.com/gitlab-org/gitlab/-/merge_requests/52658

### CODE REVIEW

File: group_hooks.go [hunk 1]
```diff
@@ -41,6 +42,7 @@ type GroupHook struct {
 	WikiPageEvents           bool       `json:"wiki_page_events"`
 	DeploymentEvents         bool       `json:"deployment_events"`
 	ReleasesEvents           bool       `json:"releases_events"`
+	SubGroupEvents           bool       `json:"subgroup_events"`
```
Comment (timofurrer) [inline group_hooks.go:45]:
```suggestion
	SubgroupEvents           bool       `json:"subgroup_events"`
```

Looking at the pattern of other names I suggest to *camel case* when *snake cased* :D

File: group_hooks.go [hunk 2]
```diff
@@ -110,6 +113,7 @@ type AddGroupHookOptions struct {
 	WikiPageEvents           *bool   `url:"wiki_page_events,omitempty"  json:"wiki_page_events,omitempty"`
 	DeploymentEvents         *bool   `url:"deployment_events,omitempty" json:"deployment_events,omitempty"`
 	ReleasesEvents           *bool   `url:"releases_events,omitempty" json:"releases_events,omitempty"`
+	SubGroupEvents           *bool   `url:"subgroup_events,omitempty" json:"subgroup_events,omitempty"`
```
Comment (timofurrer) [inline group_hooks.go:116]:
```suggestion
	SubgroupEvents           *bool   `url:"subgroup_events,omitempty" json:"subgroup_events,omitempty"`
```

File: group_hooks.go [hunk 3]
```diff
@@ -156,6 +161,7 @@ type EditGroupHookOptions struct {
 	WikiPageEvents           *bool   `url:"wiki_page_events,omitempty" json:"wiki_page_events,omitempty"`
 	DeploymentEvents         *bool   `url:"deployment_events,omitempty" json:"deployment_events,omitempty"`
 	ReleasesEvents           *bool   `url:"releases_events,omitempty" json:"releases_events,omitempty"`
+	SubGroupEvents           *bool   `url:"subgroup_events,omitempty" json:"subgroup_events,omitempty"`
```
Comment (timofurrer) [inline group_hooks.go:164]:
```suggestion
	SubgroupEvents           *bool   `url:"subgroup_events,omitempty" json:"subgroup_events,omitempty"`
```

File: group_hooks_test.go [hunk 4]
```diff
@@ -131,6 +138,7 @@ func TestGetGroupHook(t *testing.T) {
 		WikiPageEvents:           true,
 		DeploymentEvents:         true,
 		ReleasesEvents:           true,
+		SubGroupEvents:           true,
```
Comment (timofurrer) [inline group_hooks_test.go:141]:
```suggestion
		SubgroupEvents:           true,
```

File: group_hooks_test.go [hunk 5]
```diff
@@ -194,6 +205,7 @@ func TestAddGroupHook(t *testing.T) {
...
```
Comment (timofurrer) [inline group_hooks_test.go:208]:
```suggestion
		SubgroupEvents:           true,
```

File: group_hooks_test.go [hunk 6]
```diff
@@ -257,6 +272,7 @@ func TestEditGroupHook(t *testing.T) {
...
```
Comment (timofurrer) [inline group_hooks_test.go:275]:
```suggestion
		SubgroupEvents:           true,
```

File: group_hooks.go (hunk 1 above)
Comment (timofurrer) [reply group_hooks.go:45]: ... and don't get me wrong - I have the same urge to camel case to `SubGroup` ... but would favor consistency in this case 💃

**Review by timofurrer — COMMENTED** [review]
Thanks for going the extra mile 🎉

File: group_hooks.go (hunk 1 above)
Comment (svanharmelen) [reply group_hooks.go:45]: Agree with both these statements 😏 So lets keep _camel casing_ whatever the _snake cased_ names are. Even through in this case (no pun intended) I would also tent to write `SubGroup...`

Comment (jtymes) [reply group_hooks.go:45]: I was originally going to use `SubgroupEvents`, but this struct https://github.com/xanzy/go-gitlab/blob/1a0a4f28753dbb04978845336d90d402ab4b36b5/event_webhook_types.go#L894 was `SubGroupEvent`, so I was trying to be consistent with that casing. Should I change both places, or just here?

Comment (svanharmelen) [reply group_hooks.go:45]: Nice find! Please update both places then :) Thanks!

Comment (jtymes) [reply group_hooks.go]: Ok, I cleaned up those that were related to events. However, there are a few other places that use `SubGroup*` camel casing. To keep this PR focused, I left them alone for now.

**Review by svanharmelen — APPROVED** [review]
Tweaked the `SubGroup` stuff a bit.. Against my previous comment I think that in this very exceptional case I prefer to use `SubGroup` instead of `Subgroup`...

Thanks for your PR @jtymes 👍🏻

**Pull request merged**